app/src/main/java/org/android/go/sopt/ui/signup/SignUpActivity.kt

```diff
@@ -0,0 +1,107 @@
+package org.android.go.sopt.ui.signup
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.view.View
+import android.view.inputmethod.InputMethodManager
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.widget.addTextChangedListener
+import org.android.go.sopt.databinding.ActivitySignUpBinding
+import org.android.go.sopt.ui.login.LoginActivity
+import org.android.go.sopt.util.*
+
+class SignUpActivity : AppCompatActivity() {
+    private lateinit var binding: ActivitySignUpBinding
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        binding = ActivitySignUpBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        binding.root.setOnClickListener {
+            hideKeyboard()
+        }
+
+        binding.etId.addTextChangedListener {
+            if (!checkLengthOfId(it.toString())) {
+                binding.tvIdLimitError.visibility = View.VISIBLE
+            } else {
+                binding.tvIdLimitError.visibility = View.INVISIBLE
+            }
+        }
+
+        binding.etPw.addTextChangedListener {
+            if (!checkLengthOfPw(it.toString())) {
+                binding.tvPwLimitError.visibility = View.VISIBLE
+            } else {
+                binding.tvPwLimitError.visibility = View.INVISIBLE
+            }
+        }
+
+        binding.etName.addTextChangedListener {
+            if (it.toString().isEmpty()) {
+                binding.tvNameEmptyError.visibility = View.VISIBLE
+            } else {
+                binding.tvNameEmptyError.visibility = View.INVISIBLE
+            }
+        }
+
+        binding.etHobby.addTextChangedListener {
+            if (it.toString().isEmpty()) {
+                binding.tvHobbyEmptyError.visibility = View.VISIBLE
+            } else {
+                binding.tvHobbyEmptyError.visibility = View.INVISIBLE
+            }
+        }
+
+        binding.btnSignUp.setOnClickListener {
+            val inputs = getAllInputValues()
+            if(checkInputValidity(inputs)){
+                val intent = Intent(this, LoginActivity::class.java).apply {
+                    putExtra(ID_KEY, inputs[0])
+                    putExtra(PW_KEY, inputs[1])
+                    putExtra(NAME_KEY, inputs[2])
+                    putExtra(HOBBY_KEY, inputs[3])
+                }
+                setResult(RESULT_OK, intent)
+                finish()
+            }else{
+                Toast.makeText(this, INVALID_INPUT_ERROR, Toast.LENGTH_SHORT).show()
+            }
+        }
```

Comment:
```suggestion
        binding.btnSignUp.setOnClickListener {
            val inputs = getAllInputValues()
            if(checkInputValidity(inputs)){
                val intent = Intent(this, LoginActivity::class.java).apply {
                    putExtra(ID_KEY, inputs[0])
                    putExtra(PW_KEY, inputs[1])
                    putExtra(NAME_KEY, inputs[2])
                    putExtra(HOBBY_KEY, inputs[3])
                }
                setResult(RESULT_OK, intent)
                finish()
                return@setOnClickListener
            }
            Toast.makeText(this, INVALID_INPUT_ERROR, Toast.LENGTH_SHORT).show()
        }
```
사람마다 코드 스타일이 다르지만 early return 패턴으로 작성할 수도 있습니다!